### PR TITLE
Socker board impl

### PR DESCRIPTION
**Pull request by Ilia-tod29**
Using sockets so the current player can see the board of the enemy player.

### CODE REVIEW

File: server/src/index.js [hunk 1]
```diff
@@ -22,16 +24,38 @@ async function start() {
...
+
+    socket.on('gameMessage', message => {
+      if (message.playOn === true) {
+        playersReadyMap.set(roomNo, playersReadyMap.get(roomNo) + 1);
+        console.log(playersReadyMap.get(roomNo));
+      } else if (message.playOn === false) {
+        if (playersReadyMap.get(roomNo) !== 0) {
+          playersReadyMap.set(roomNo, playersReadyMap.get(roomNo) - 1);
+        }
+        console.log(playersReadyMap.get(roomNo));
```
Comment (GritzMaze) [inline server/src/index.js:51]: this console logs are not needed

File: server/src/index.js [hunk 2]
```diff
@@ -22,16 +24,38 @@ async function start() {
...
+      io.to(roomNo).emit('receiveChatMessage', {
         userId: socket.id,
         message: message
       });
     });
+
+    socket.on('gameMessage', message => {
+      if (message.playOn === true) {
+        playersReadyMap.set(roomNo, playersReadyMap.get(roomNo) + 1);
+        console.log(playersReadyMap.get(roomNo));
```
Comment (GritzMaze) [inline server/src/index.js:46]: this as well

File: client/src/components/game-board/game-board.js [hunk 3]
```diff
@@ -370,18 +381,20 @@ class GameBoard extends HTMLElement {
...
-  attrbuteChangedCallback(prop, oldValue, newValue) {
-    if (prop === 'isready') {
-      console.log('Attribute changed');
-    }
-  }
+  // attrbuteChangedCallback(prop, oldValue, newValue) {
+  //   if (prop === 'isready') {
+  //     console.log('Attribute changed');
+  //   }
+  // }
```
Comment (GritzMaze) [inline client/src/components/game-board/game-board.js]: delete

File: client/src/components/game-board/game-board.js [hunk 4]
```diff
@@ -370,18 +381,20 @@ class GameBoard extends HTMLElement {
...
+      this.shuffle();
+    }
   };
 
-  static get observedAttributes() {
-    return ['isready'];
-  }
+  // static get observedAttributes() {
+  //   return ['isready'];
+  // }
```
Comment (GritzMaze) [inline client/src/components/game-board/game-board.js]: delete

File: client/src/components/game-board/game-board.js [hunk 5]
```diff
@@ -103,6 +97,12 @@ class GameBoard extends HTMLElement {
     this.loadGame();
     // Starting the timer
     this.setAttribute('isready', true);
+
+    socketService.emitMessage('gameMessage', {playOn: true});
+    // socketService.emitMessage('gameMessage', JSON.stringify(this.#_shadowRoot));
+    // console.log(this);
```
Comment (GritzMaze) [inline client/src/components/game-board/game-board.js]: delete

File: client/src/app.js [hunk 6]
```diff
@@ -146,7 +213,7 @@ export default class AppComponent extends HTMLElement {
   }
 
   connectedCallback() {
-    this.listenForGame();
+    // this.listenForGame();
```
Comment (GritzMaze) [inline client/src/app.js]: delete

File: client/src/app.js [hunk 7]
```diff
@@ -55,40 +59,103 @@ export default class AppComponent extends HTMLElement {
...
+  //   const element = this.#_shadowRoot.getElementById("playerA");
+
+  //   if(!element) {
+  //       window.setTimeout(this.listenForGame, 1000);
+
+  //       return;
+  //   }
+
+  //   observer.observe(this.#_shadowRoot.getElementById("playerA"), { attributes: true });
+  // }
```
Comment (GritzMaze) [inline client/src/app.js]: delete

**Review by GritzMaze — APPROVED** [review]

**Review by boqnd — APPROVED** [review]

**Pull request merged**